### PR TITLE
Added Capability to Parse Nest Json Configs

### PR DESCRIPTION
**Pull request by nickuncaged1201**
Added support to parsed nest json configs. This is needed to configure file schema transforms properly.

### CODE REVIEW

File: syndeo-template/src/main/java/com/google/cloud/syndeo/SyndeoTemplate.java [hunk 1]
```diff
@@ -189,11 +193,55 @@ public static ConfiguredSchemaTransform buildFromJsonConfig(JsonNode transformCo
                                         ? fieldNode.asText()
                                         : fieldNode.isArray()
                                             ? new ObjectMapper().convertValue(fieldNode, List.class)
-                                            : new ObjectMapper().convertValue(fieldNode, Map.class))
```
Comment (pabloem) [inline syndeo-template/src/main/java/com/google/cloud/syndeo/SyndeoTemplate.java:192]: you should address this later on, but it is also possible taht the field is a `Map` type. You will need to pass in the whole Field object (from [here](https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/714/files#diff-a467001c5b58c9a73fafb006c786361ca4dad3e61c38bc8132503b3034b24b92R180)) to check if it's of type Map or type Row.

File: syndeo-template/src/main/java/com/google/cloud/syndeo/SyndeoTemplate.java [hunk 2]
```diff
@@ -189,11 +193,55 @@ public static ConfiguredSchemaTransform buildFromJsonConfig(JsonNode transformCo
...
+    ConfiguredSchemaTransform t = new ProviderUtil.TransformSpec(
             transformConfig.get("urn").asText(), configurationParameters)
         .toProto();
+
+    LOG.warn("configured schema transfrm " + t.getConfigurationValues().toString());
+    // LOG.warn("fiel " + t.getConfigurationValues().)
+    return t;
+  }
+
+  private static Row convertJsonNodeToRow(JsonNode node) {
```
Comment (pabloem) [inline syndeo-template/src/main/java/com/google/cloud/syndeo/SyndeoTemplate.java:210]: here you can probably get the `Field` object, and get the schema from it via `Field.getType().getRowSchema()` - that way you don't have to infer the field automatically.

**Comment by nickuncaged1201** [comment]
r: @Dippatel98 @pabloem

File: syndeo-template/src/main/java/com/google/cloud/syndeo/SyndeoTemplate.java [hunk 3]
```diff
@@ -173,29 +178,60 @@ public static ConfiguredSchemaTransform buildFromJsonConfig(JsonNode transformCo
...
+                        ? fieldNode.asLong()
+                        : !fieldNode.isContainerNode()
+                            ? fieldNode.asText()
+                            : fieldNode.isArray()
+                                ? new ObjectMapper().convertValue(fieldNode, List.class)
+                                : field.getType().getTypeName() == TypeName.MAP
+                                    ? new ObjectMapper().convertValue(fieldNode, Map.class)
+                                    : field.getType().getTypeName() == TypeName.ROW
+                                        ? convertJsonNodeToRow(fieldPair)
+                                        : new ObjectMapper().convertValue(fieldNode, Object.class);
```
Comment (bvolpato) [inline syndeo-template/src/main/java/com/google/cloud/syndeo/SyndeoTemplate.java:218]: I think if () and else if () blocks would look much cleaner and less error prone here

**Review by bvolpato — DISMISSED** [review]
LGTM, but left a style comment if you want to address if before submitting

Thanks Nick

**Comment by Dippatel98** [comment]
LGTM

**Comment by bvolpato** [comment]
This PR got messed up -- can you rebase?

**Comment by nickuncaged1201** [comment]
> This PR got messed up -- can you rebase?

Done! Please merge if possible

File: syndeo-template/src/main/java/com/google/cloud/syndeo/SyndeoTemplate.java [hunk 4]
```diff
@@ -173,29 +178,74 @@ public static ConfiguredSchemaTransform buildFromJsonConfig(JsonNode transformCo
...
+    //                     : !fieldNode.isContainerNode()
+    //                         ? fieldNode.asText()
+    //                         : fieldNode.isArray()
+    //                             ? new ObjectMapper().convertValue(fieldNode, List.class)
+    //                             : field.getType().getTypeName() == TypeName.MAP
+    //                                 ? new ObjectMapper().convertValue(fieldNode, Map.class)
+    //                                 : field.getType().getTypeName() == TypeName.ROW
+    //                                     ? convertJsonNodeToRow(fieldPair)
+    //                                     : new ObjectMapper().convertValue(fieldNode,
+    // Object.class);
```
Comment (bvolpato) [inline syndeo-template/src/main/java/com/google/cloud/syndeo/SyndeoTemplate.java:218]: Better to clean this up

**Review by bvolpato — APPROVED** [review]

**Review by bvolpato — APPROVED** [review]

**Pull request merged**